gaia/geo/gdal_functions.py

```diff
@@ -222,11 +271,25 @@ def world_to_pixel(geoMatrix, x, y):
     # create output raster
     raster_band = raster_input.GetRasterBand(1)
     output_driver = gdal.GetDriverByName('MEM')
+
+    # In the event we have multispectral images, shift the shape dimesions we are after,
+    # since position 0 will be the number of bands
+    clip_shp_0 = clip.shape[0]
+    clip_shp_1 = clip.shape[1]
+    if clip.ndim > 2:
+        clip_shp_0 = clip.shape[1]
+        clip_shp_1 = clip.shape[2]
+
     output_dataset = output_driver.Create(
-        '', clip.shape[1], clip.shape[0],
+        '', clip_shp_1, clip_shp_0,
         raster_input.RasterCount, raster_band.DataType)
     output_dataset.SetGeoTransform(geo_trans)
-    output_dataset.SetProjection(raster_input.GetProjection())
+    output_dataset.SetProjection(gdal_get_projection(raster_input))
+
+    #Copy RPC data from src to dst
+    RPC = src_image.GetMetadata('RPC')
```

Comment:
@aashish24 mentioned trying to copy all of the metadata fields.  I wonder if you can use `GetMetadataDomainList()` to get the keys and then loop through calling `GetMetadata` on each?